astropy/io/fits/fitstime.py

```diff
@@ -0,0 +1,251 @@
+        """
+        Set the global time reference frame attributes.
+
+        Parameters
+        ----------
+        key : str
+            FITS global time reference frame keyword.
+        value : int, str, list
+            value associated with specified keyword.
+        comment : str
+            comment associated with specified keyword.
+        """
+        if key in self.TIME_KEYWORDS:
+            self.global_info[self.TIME_KEYWORDS[key]] = value
+        else:
+            raise ValueError('Illegal global time keyword.')
+
+    def set_column_override(self, key, value, comment=None):
+        """
+        Set the time column specific override attributes.
+
+        Parameters
+        ----------
+        key : str
+            FITS time column specific keyword.
+        value : int, str, list
+            value associated with specified keyword.
+        comment : str
+            comment associated with specified keyword.
+        """
+        if self.is_time_column_keyword(key):
+            idx = int(key[5:])
+            self.time_columns[idx][self.COLUMN_TIME_KEYWORDS[key[:5]]] = value
+        else:
+            raise ValueError('Illegal column-specific time keyword.')
+
+    def convert_time_columns(self, table):
+        """
+        Convert time columns to Astropy Time columns.
+
+        Parameters
+        ----------
+        table : astropy.table.Table
+            The table whose time columns are to be converted.
+        """
+        for idx, time_col in self.time_columns.items():
+            time_colname = table.colnames[idx - 1]
+            if self.astropy_file is not None:
+                table[time_colname] = Time(table[time_colname][...,0], table[time_colname][...,1],
+                                           format='jd', scale=time_col['scale'].lower())
+                try:
+                    if time_col['pos'] == 'TOPOCENTER':
+                        table[time_colname].location = EarthLocation(self.global_info['loc_x'],
+                                                                     self.global_info['loc_y'],
+                                                                     self.global_info['loc_z'], 
+                                                                     unit='m')
+                except:
+                    pass
+            else:
+                # Still have to complete this to read FITS files not written by astropy
+                pass
+
+    def read_time(self, hdr, table):
+        """
+        Set the time coordinate state of a FITS Binary Table.
+
+        Parameters
+        ----------
+        hdr : `~astropy.io.fits.header.Header`
+            FITS Header
+        table : astropy.table.Table
+            The table whose time columns are to be read as Time
+        """
+        for key, value, comment in hdr.cards:
+            if (key.upper() in TIME_KEYWORDS):
+
+                self.set_global_time(key, value, comment)
+                hdr.remove(key)
```

Comment:
Do you really need to remove the key? The astropy table shall not have this keyword, but do we need to delete it from the original table? I don't know when the table.meta is populated, but if it is already at this point, could be delete it from there? I realize that I'm probably asking a very simple question that does not have an easy answer.

My worry is a situation like this (pseudo code):
```
hdus = fits.open('myfile.fits')
table = Table.read(hdus[1])
... do stuff with table ...
now, do stuff with hdu, e.g.
hdus[1].add_column_and_other_information
hdus.save()
```
In that case, the hdus written back won't be valid fits time any longer. Now, I did not trace all the details of the order of how this code is executed, so I'm asking: Is this "hdr" object where keywords are removed a temporary copy or do we have to worry that users continue to use it after we've deleted keywords from it?